coreneuron/mechanism/eion.cpp

```diff
@@ -233,7 +233,7 @@ ion_style("name_ion", [c_style, e_style, einit, eadvance, cinit])
  and models.
 */
 
-#define iontype ppd[0] /* how _AMBIGUOUS is to be handled */
+#define iontype ppd[_iml] /* how _AMBIGUOUS is to be handled */
```

Comment:
@nrnhines : this looks reasonable? I wonder if it's possible to update any existing test to catch this (Ioannis needed quite some efforts to track this / reach here).

Reply:
That looks reasonable. It is more analogous to the NEURON definition.
```
#define iontype ppd[i][0].i    /* how _AMBIGUOUS is to be handled */
```
I can think of several ways to force ppd to be different in different sections. In NEURON one would only need to have different ```ion_style``` args so that the type was STATE, ASSIGNED, PARAMETER, etc.
That should get copied to CoreNEURON. If one considers ```nrn_cur_ion``` at
```
        if (iontype & 0100) {
            erev = nrn_nernst(conci, conco, charge, celsius);
        }
```
Then one could have different sections where iontype&0100 is different and the ones that are true would calculate the erev based on conci, etc.  Of course one would need a model that had time variation of conci
to test that vector.record of erev (or just the final erev on return)  from coreneuron psolve was the same as the neuron psolve.

Reply:
Ok. I mentioned it in case it's easy to adapt any existing test. Anyway, this PR can be merged to have the main issued fixed in 8.1 release for the paper.

Reply:
I think we can merge this now and include it in the paper release so that the results between NEURON and CoreNEURON direct mode are the same and I'll create a ticket to add a test related to this